mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MessageCodec.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.mesos.rx.java;
+
+import org.jetbrains.annotations.NotNull;
+
+public interface MessageCodec<T> {
```

Comment:
That makes sense. I'm not quite sure what you mean by `encode` and `decode` producing different `Observable`s but I'll get started on updating the code.